src/NUnitFramework/framework/Internal/TestCaseParameters.cs

```diff
@@ -80,5 +80,12 @@ public TestCaseParameters(ITestCaseData data) : base(data)
         public bool HasExpectedResult { get; set; }
 
         #endregion
+
+        /// <summary>
+        /// Get or set the type arguments for a generic test method.
+        /// If not set explicitly, the generic types will be inferred
+        /// based on the test case parameters.
+        /// </summary>
+        public Type[]? TypeArgs { get; set; } = null;
```

Comment:
I was surprised at just _how_ little was required to make it work for TestCaseSource after TestCase was done. If I had made this public originally instead of private then it would've "just worked"